assets/source/setup-guide/app/components/Account/Connection.js

```diff
@@ -0,0 +1,203 @@
+							'Are you sure you want to disconnect this account?',
+							'pinterest-for-woocommerce'
+						) }
+					</p>
+					<div className="woocommerce-setup-guide__step-modal__buttons">
+						<Button
+							isDestructive
+							isSecondary
+							onClick={ handleDisconnectAccount }
+						>
+							{ __(
+								"Yes, I'm sure",
+								'pinterest-for-woocommerce'
+							) }
+						</Button>
+						<Button isTertiary onClick={ closeConfirmationModal }>
+							{ __( 'Cancel', 'pinterest-for-woocommerce' ) }
+						</Button>
+					</div>
+				</div>
+			</Modal>
+		);
+	};
+
+	const handleDisconnectAccount = async () => {
+		closeConfirmationModal();
+
+		const result = await apiFetch( {
+			path:
+				wcSettings.pinterest_for_woocommerce.apiRoute +
+				'/auth_disconnect',
+			method: 'POST',
+		} );
+
+		if ( ! result.disconnected ) {
+			createNotice(
+				'error',
+				__(
+					'There was a problem while trying to disconnect.',
+					'pinterest-for-woocommerce'
+				)
+			);
+		} else {
+			setIsConnected( false );
+
+			// Force reload WC admin page to initiate the relevant dependencies of the Dashboard page.
+			const path = getNewPath( {}, '/pinterest/landing', {} );
+
+			window.location = new URL(
+				decodeEntities( wcSettings.adminUrl + path )
+			);
+		}
+	};
+
+	return (
+		<CardBody size="large">
+			{ isConnected === true ? ( // eslint-disable-line no-nested-ternary --- Code is reasonable readable
+				<Flex direction="row" className="connection-info">
+					{ accountData?.id ? (
+						<>
+							<FlexItem className="logo">
+								<PinterestLogo />
+							</FlexItem>
+
+							<FlexBlock className="account-label">
+								<Text variant="body">
+									{ accountData.username }
+
+									<span className="account-type">
+										{ ' - (' }
+										{ accountData.is_partner
+											? __(
+													'Business account',
+													'pinterest-for-woocommerce'
+											  )
+											: __(
+													'Personal account',
+													'pinterest-for-woocommerce'
+											  ) }
+										{ ')' }
+									</span>
+								</Text>
+							</FlexBlock>
+
+							<FlexItem>
+								<Button
+									isLink
+									isDestructive
+									onClick={ openConfirmationModal }
+								>
+									{ __(
+										'Disconnect',
+										'pinterest-for-woocommerce'
+									) }
+								</Button>
+							</FlexItem>
+						</>
+					) : (
+						<Spinner />
+					) }
+				</Flex>
+			) : isConnected === false ? (
+				<Flex direction="row" className="connection-info">
+					<FlexItem className="logo">
+						<PinterestLogo />
+					</FlexItem>
+
+					<FlexBlock>
+						<Text variant="subtitle">
+							{ __(
+								'Connect your Pinterest Account',
+								'pinterest-for-woocommerce'
+							) }
+						</Text>
+					</FlexBlock>
+
+					<FlexItem>
+						<Button
+							isSecondary
+							href={ decodeEntities(
+								wcSettings.pinterest_for_woocommerce
+									.serviceLoginUrl
+							) }
```

Comment:
No need to encode and decode `serviceLoginUrl`.

Reply:
Discussed [here](https://github.com/woocommerce/pinterest-for-woocommerce/pull/213#discussion_r720090383)